vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -761,6 +791,10 @@ def get_rope(
     if key in _ROPE_DICT:
         return _ROPE_DICT[key]
     if rope_scaling is None:
+        if max_position == 131072:
+            # Note(simon): this is a special case for a model that doesn't 
+            # supply rope_scaling. We should remove this once the model is updated.
+            RotaryEmbedding = ExtendedRotaryEmbedding
```

Comment:
Why don't you just do 
```
rotary_emb = ExtendedRotaryEmbedding(head_size, rotary_dim, max_position, base,
                                     is_neox_style, dtype)
```

This could break speculative decoding, for instance, since you may want to use different RoPE impl for draft and target models.

Also, the key in _ROPE_DICT should probably indicate a model identifier of some kind to avoid a similar bug.

Reply:
Good point.

Reply:
Also, is there something else we can key off of here to make sure to avoid false positives?

Reply:
I'm going to do the change but not adding in model id because that's pretty intrusive and won't be needed after the proper fix. Do you think that's okay?

Reply:
I don't know how common this particular `max_position` is, probably not that common, but this would enable it for them. Could you also key on `base`?